src/jit/codegenarmarch.cpp

```diff
@@ -1617,11 +1766,29 @@ void CodeGen::genCallInstruction(GenTreeCall* call)
         }
         else
         {
-            regNumber argReg = curArgTabEntry->regNum;
             genConsumeReg(argNode);
```

Comment:
Yes. I think you are correct in that, if we always expect a `GT_PUTARG_SPLIT`, then it is sufficient to ensure that any spill or unspill is handled as needed.  But I think it would be best to handle this in a consistent way - e.g. with something like a `genConsumePutArgSplit`.